code/script.js

```diff
@@ -231,95 +235,135 @@ const setSecret = () => {
+      keepAccessoriesOther(category, value)
+      alert(`Yes, the person wears ${value}! Keep all people that wears ${value}`)
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      removeAccessoriesOther(category, value)
+      alert(`No, the person doesn't wear ${value}! Remove all people that wears ${value}`)
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      keepAccessoriesOther(category, value)
+      alert(`Yes, the person is a ${value}! Keep all people that are ${value}s`)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      removeAccessoriesOther(category, value)
+      alert(`No, the person is not a ${value}! Remove all people that are ${value}s`)
+    }
+  } else if (category === 'hair') {
+    if (keep) {
+      keepHairEyes(category, value)
+      alert(`Yes, the person is a ${value} ${category}! Keep all people with ${value} ${category}`)
+    } else {
+      removeHairEyes(category, value)
+      alert(`No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`)
+    }
+  } else { // if (category === 'eyes)
+    if (keep) {
+      keepHairEyes(category, value)
+      alert(`Yes, the person is a ${value} ${category}! Keep all people with ${value} ${category}`)
+    } else {
+      removeHairEyes(category, value)
+      alert(`No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`)
     }
   }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  // Invoke function to redraw the board with the remaining people
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking guess, the player first have to confirm that they want to make a guess
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const confirmation = confirm(`Are you sure you want to make a guess on ${personToConfirm}?`)
+  if (confirmation === true) {
```

Comment:
I learned that these 2 lines of codes could be written in one line without variable (const) needed. It would look like this: 

if (confirm(`Are you sure you want to make a guess on ${personToConfirm}?`) === true) {

Although, both ways works fine of course:-)